### PR TITLE
Add the persistent utils.delay

### PR DESCRIPTION
**Pull request by vlegoff**
#### Brief overview of PR changes/additions

This PR allows a `persistent` keyword argument to be set in the `evennia.utils.delay` function.  This allows to create persistent tasks that are stored in the `ServerConfig` and will be executed even after a reload or restart.

#### Motivation for adding to Evennia

Motivations for this pull request are detailed in the [issue #1341](https://github.com/evennia/evennia/issues/1341).

#### Notes

- I didn't include tests.  How can we test it?  Can we track deferred and force them to execute somehow?
- It could add a bit more security.  If the object is destroyed in between, I'm not sure what happens.
- It uses a "tweak" to allow saving of instance methods.  When an instance method is detected in a callback, it will save a tuple (the object, the method name) and will restore it in loading the configuration.  I was wondering if perhaps it could be useful to add it in the serializer itself, but maybe not practical.

#### Other info (issues closed, discussion etc)

- (Issue #1341)[https://github.com/evennia/evennia/issues/1341]

### CODE REVIEW

**Comment by Griatch** [comment]
@v-legoff This looks good wi. I am not sure about the need for the `utils.persistent` module though. I'd be interested in hearing your arguments for having it over letting the `delay()` function handle things.

**Comment by Griatch** [comment]
After discussion in IRC, it was agreed the persistent-class better be reworked into a new `TaskHandler` to  go into `evennia/scripts/taskhandler.py`. The `utils.delay(... persistent=True)` then just becomes a wrapper for calling this handler, which also opens up the handler for other uses by other systems.

**Comment by vlegoff** [comment]
@Griatch: I have moved the persistent mechanism in the task handler, which is called for persistent and non-persistent tasks by `utils.delay` now.

File: evennia/scripts/taskhandler.py [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    def create_delays(self):
+        """Create the delayed tasks for the persistent tasks.
+
+        Note:
+            This method should be automatically called when Evennia starts.
+
+        """
+        now = datetime.now()
+        for task_id, (date, callbac, args, kwargs) in self.tasks.items():
+            seconds = (date - now).total_seconds()
```
Comment (Griatch) [inline evennia/scripts/taskhandler.py]: Minor quibble: The standard way to make sure a value is bigger or equal to zero is to do `seconds = max(0, (data - now).total_seconds())`. You can then remove the `if` statement.

**Comment by vlegoff** [comment]
Fixed!  Thganks for the trick, I learned something.

**Pull request merged**